source/matrix_free/task_info.cc

```diff
@@ -413,6 +413,13 @@ namespace internal
               root->wait_for_all();
               root->destroy(*root);
             }
+          else if(scheme == partition_partition)
+            {
+              // catch the case of empty partition list where we still finish
+              // the exchange of ghost data
+              funct.vector_update_ghosts_finish();
+              funct.vector_compress_start();
+            }
```

Comment:
The problem was that in the other code we would not even enter the code that set up some of the communication because it sits inside a loop over cell blocks - without cells, there is no loop iteration that could execute the code.